packages/cms-data-sync/test/index.test.ts

```diff
@@ -0,0 +1,120 @@
+import * as contentfulManagement from 'contentful-management';
+import { runMigrations } from '../src/index';
+import { BLUE_COLOR, RED_COLOR } from '../src/utils';
+
+import { migrateTeams } from '../src/teams/teams.data-migration';
+import { migrateExternalAuthors } from '../src/external-authors/external-authors.data-migration';
+import { migrateCalendars } from '../src/calendars/calendars.data-migration';
+
+jest.mock('contentful-management');
+
+var mockMigrateTeams: jest.MockedFunction<typeof migrateTeams>;
+
+jest.mock('../src/teams/teams.data-migration', () => {
+  mockMigrateTeams = jest.fn().mockReturnValue({});
+  return {
+    ...jest.requireActual('../src/teams/teams.data-migration'),
+    migrateTeams: mockMigrateTeams,
+  };
+});
+
+var mockMigrateExternalAuthors: jest.MockedFunction<
+  typeof migrateExternalAuthors
+>;
+
+jest.mock('../src/external-authors/external-authors.data-migration', () => {
+  mockMigrateExternalAuthors = jest.fn().mockReturnValue({});
+  return {
+    ...jest.requireActual(
+      '../src/external-authors/external-authors.data-migration',
+    ),
+    migrateExternalAuthors: mockMigrateExternalAuthors,
+  };
+});
+
+var mockMigrateCalendars: jest.MockedFunction<typeof migrateCalendars>;
+
+jest.mock('../src/calendars/calendars.data-migration', () => {
+  mockMigrateCalendars = jest.fn().mockReturnValue({});
+  return {
+    ...jest.requireActual('../src/calendars/calendars.data-migration'),
+    migrateCalendars: mockMigrateCalendars,
+  };
+});
+
+const mockContentfulManagement = contentfulManagement as jest.Mocked<
+  typeof contentfulManagement
+>;
+
+const webhookMock = { update: jest.fn(), active: true };
+const spaceMock = {
+  getWebhook: jest.fn().mockResolvedValue(webhookMock),
+};
+const getSpaceFn = jest.fn().mockResolvedValue(spaceMock);
+mockContentfulManagement.createClient.mockReturnValue({
+  getSpace: getSpaceFn,
+} as any as jest.Mocked<contentfulManagement.PlainClientAPI>);
+
+describe('Migrations', () => {
+  const consoleLogRef = console.log;
+
+  beforeEach(() => {
+    console.log = jest.fn();
+    process.env.CONTENTFUL_ENV_ID = 'crn-env-id';
+    process.env.VERBOSE_DATA_SYNC = 'true';
+    jest.clearAllMocks();
+  });
+
+  afterAll(() => {
+    console.log = consoleLogRef;
+  });
+
+  it('deactivates webhook and activates it again after running the migrations', async () => {
+    await runMigrations();
+
+    expect(spaceMock.getWebhook).toHaveBeenCalledWith('crn-env-id-webhook');
+    expect(webhookMock.update).toHaveBeenCalled();
+    expect(console.log).toHaveBeenNthCalledWith(
```

Comment:
I take it this is your way of testing that the webhook was deactivated as technically we are unable to assert that webhookMock.active was set to false since immediately after running it is changed to true? I think we should try and test that better, even if this is a throwaway code. 

One way would be to create setter function on `webhookMock` here https://github.com/yldio/asap-hub/blob/06a948055b7c06d252691911f8a9a430a1027fe7/packages/cms-data-sync/test/index.test.ts#L49

That way you could set some global variable once active is set to false and then assert that that global var was set correctly, what do you think?

Reply:
added here [review: setter active test](https://github.com/yldio/asap-hub/pull/2935/commits/ea92fda801e831b8592bfaa76fdcf42335361f8c)